tic_tac_toe/game.py

```diff
@@ -87,18 +124,49 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
-
+    if not _position_is_valid(position): 
+        raise InvalidMovement('Position out of range.')
+    if game['winner'] or _board_is_full(game['board']): #game is over
+        raise InvalidMovement('Game is over.')    
+    if not _position_is_empty_in_board(position, game['board']):
+        raise InvalidMovement('Position already taken.')
+    if not player == game['next_turn']:
+        raise InvalidMovement('"%s" moves next'%(game['next_turn']))
+    
+    
+    i = position[0]
+    j = position[1]
+    game['board'][i][j] = player
+    
+    if player == game['player1']:
+        game['next_turn'] = game['player2']
+    else:
+        game['next_turn'] = game['player1']
+    
+    if _check_winning_combinations(game['board'], player):
+        game['winner'] = player
+        raise GameOver('"%s" wins!'%(player))
+    if _board_is_full(game['board']): #game is tied
+        raise GameOver('Game is tied!')
 
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
+    
+    ret = []
+    dashes = "--------------\n"
+    for row in game['board']:
+        first = row[0]
+        second = row[1]
+        third = row[2]
+        row_str = "%(first)s  |  %(second)s  |  %(third)s\n"%{'first': first, 'second': second, 'third': third}
```

Comment:
Again, you're setting variables to just use them once when it would be simpler to just put those values in directly. Also, using `.format()` is preferred to `%` for string formatting. Example of how doing this would simplify things:
```
for row in game['board']:
   ret.append("{}  |  {}  |  {}\n".format(row[0], row[1], row[2])")
```
Here's a page that will show you the different ways to format using both the old and new methods, along with features that are exclusive to the new method: [PyFormat](https://pyformat.info/)